mala/targets/calculation_helpers.py

```diff
@@ -62,8 +62,25 @@ def fermi_function(energy, fermi_energy, temperature):
         Value of the Fermi function.
 
     """
-    return 1.0 / (1.0 + np.exp((energy - fermi_energy) /
-                               (kB * temperature)))
+    exponent = (energy - fermi_energy) / (kB * temperature)
+
+    # Maximum exponent that will not result in inf when performing
+    # np.exp() (for double values only).
+    # Very large exponents result in very large values, and if we don't check
+    # for overflow, infinity. Cutting this back to the maximum float value
+    # (~ 1e308) is more then enough, since we divide by the exponent, and
+    # therefore get zero (or machine precision within zero) either way.
+    # Adding this check removes overflow warnings.
+    # The overhead for inference should be minimal.
+    # Since this function works both for arrays and scalars,
+    # we have to check which maximum to use.
+    if type(energy) == np.ndarray:
```

Comment:
`isinstance(energy, np.ndarray)` is the more general way to do this (even though we won't expect to get passed ndarray subclasses here).